src/Microsoft.Net.Http.Headers/MediaTypeHeaderValue.cs

```diff
@@ -441,12 +441,24 @@ private static int GetMediaTypeLength(string input, int startIndex, out MediaTyp
             return current - startIndex;
         }
 
-        private static int GetMediaTypeExpressionLength(string input, int startIndex, out string mediaType)
+        internal static int ParseMediaTypeExpresion<T, K>(
```

Comment:
Don't want to change the visibility more than needed unless we discuss otherwise and do a comprehensive change on the whole package, the public API for this is MediaTypeParser